TestCopyRange fails on Windows 2025

### What happened?

Locally, on Windows 11, and on the windows-2025 GitHub runner (go 1.23 and 1.24), the `TestCopyRange` test is failing with `The request is not supported.`

To fix this, we need to change:

```go
err == syscall.ENOTSUP
```
to
```go
errors.Is(err, errors.ErrUnsupported)
```
a PR is forthcoming.

See also https://forum.syncthing.net/t/the-request-is-not-supported/24462

### Syncthing version

v2.0.0-rc.20

### Platform & operating system

Windows 11

### Browser version

n/a

### Relevant log output

```shell
=== RUN   TestCopyRange
=== RUN   TestCopyRange/tmp
=== RUN   TestCopyRange/tmp/duplicate_extents
=== RUN   TestCopyRange/tmp/duplicate_extents/append_to_end
    filesystem_copy_range_test.go:339: The request is not supported.
--- FAIL: TestCopyRange/tmp/duplicate_extents/append_to_end (0.02s)
=== RUN   TestCopyRange/tmp/duplicate_extents/append_to_end_offsets_at_start
    filesystem_copy_range_test.go:339: The request is not supported.
--- FAIL: TestCopyRange/tmp/duplicate_extents/append_to_end_offsets_at_start (0.02s)
=== RUN   TestCopyRange/tmp/duplicate_extents/overwrite_part_of_destination_region
    filesystem_copy_range_test.go:339: The request is not supported.
--- FAIL: TestCopyRange/tmp/duplicate_extents/overwrite_part_of_destination_region (0.02s)
=== RUN   TestCopyRange/tmp/duplicate_extents/overwrite_all_of_destination
    filesystem_copy_range_test.go:339: The request is not supported.
--- FAIL: TestCopyRange/tmp/duplicate_extents/overwrite_all_of_destination (0.02s)
=== RUN   TestCopyRange/tmp/duplicate_extents/overwrite_part_of_destination
    filesystem_copy_range_test.go:339: The request is not supported.
--- FAIL: TestCopyRange/tmp/duplicate_extents/overwrite_part_of_destination (0.02s)
=== RUN   TestCopyRange/tmp/duplicate_extents/destination_gets_expanded_as_it_is_being_wri
```

--- a/lib/fs/filesystem_copy_range_test.go
+++ b/lib/fs/filesystem_copy_range_test.go
@@ -8,6 +8,7 @@ package fs
 
 import (
 	"bytes"
+	"errors"
 	"io"
 	"math/rand"
 	"os"
@@ -329,7 +330,7 @@ func TestCopyRange(tttt *testing.T) {
 								t.Fatal("dst file is not a basic file")
 							}
 							if err := impl(srcBasic, dstBasic, testCase.srcOffset, testCase.dstOffset, testCase.copySize); err != nil {
-								if err == syscall.ENOTSUP {
+								if errors.Is(err, errors.ErrUnsupported) {
 									// Test runner can adjust directory in which to run the tests, that allow broader tests.
 									t.Skip("Not supported on the current filesystem, set STFSTESTPATH env var.")
 								}

